### DISCUSSION TITLE
http2 protocol

### DISCUSSION THREAD

**Discussion by JMo911** [mswjs/msw#1352] [Q&A]
Is it possible to config msw to send h2 protocol responses instead of http/1.1? See here, when running without mocks all of the calls follow h2. As soon as I enable msw mocks it comes back as http/1.1.

![Screen Shot 2022-07-25 at 2 09 38 PM](https://user-images.githubusercontent.com/32887744/180855734-8a0491a8-e6fb-4248-ae07-d37e1b27ea29.png)

**Comment by kettanaito** [comment]
Hey, @JMo911. Thanks for raising this.

So, this may be a little confusing, allow me to clarify. When you perform a request over HTTP2, this is what happens:

1. Network records the HTTP2 request.
2. The request is captured by the worker. If the worker responds to the request (or performs another request while the former is pending), it will be recorded as HTTP/1.1. Note that this **is not related to your original HTTP2 request**.
3. You see the worker's HTTP/1.1 request resolved in the Network.
4. Then, you see your original HTTP2 request resolved in the Network.

When you mock an HTTP2 request, your app will receive the mocked response to that original HTTP2 request, regardless of what you see in the Network. [This](https://stackoverflow.com/a/54515338/2754939) may be a good answer to read. 

Do you experience some difficulty due to this behavior? Or is it just something you've noticed in the Network? It shouldn't affect your application in any way. We also can't control this, it's how the workers behave.
